python/ray/experimental/client/common.py

```diff
@@ -27,12 +38,64 @@ def __call__(self, *args, **kwargs):
                         "Use {self._name}.remote method instead")
 
     def remote(self, *args, **kwargs):
-        return ray.call_remote(self, *args, **kwargs)
+        return ray.call_remote(self, ray_client_pb2.ClientTask.FUNCTION, *args,
+                               **kwargs)
 
     def __repr__(self):
         return "ClientRemoteFunc(%s, %s)" % (self._name, self.id)
 
 
+class ClientActorClass:
+    def __init__(self, actor_cls):
+        self.actor_cls = actor_cls
+        self._name = actor_cls.__name__
+
+    def __call__(self, *args, **kwargs):
+        raise TypeError(f"Remote actor cannot be instantiated directly. "
+                        "Use {self._name}.remote() instead")
+
+    def remote(self, *args, **kwargs):
+        # Actually instantiate the actor
+        ref = ray.call_remote(self, ray_client_pb2.ClientTask.ACTOR, *args,
+                              **kwargs)
+        return ClientActorHandle(ref, self)
+
+    def __repr__(self):
+        return "ClientRemoteActor(%s, %s)" % (self._name, self.id)
+
+    def __getattr__(self, key):
+        raise NotImplementedError("static methods")
+
+
+class ClientActorHandle:
+    def __init__(self, actor_id: ClientActorRef,
+                 actor_class: ClientActorClass):
+        self.actor_id = actor_id
+        self.actor_class = actor_class
+
+    def __getattr__(self, key):
+        return ClientRemoteMethod(self, key)
```

Comment:
It seems __getattr__ is intercepting too much here, e.g., if you try to pickle this handle it ends up calling __getattr__ which returns a remote method.